plugin/pkg/scheduler/factory/factory.go

```diff
@@ -168,14 +169,41 @@ func (factory *ConfigFactory) createMinionLW() *cache.ListWatch {
 	}
 }
 
-// pollMinions lists all minions and returns an enumerator for cache.Poller.
+// pollMinions lists all minions and filter out unhealthy ones, then returns
+// an enumerator for cache.Poller.
 func (factory *ConfigFactory) pollMinions() (cache.Enumerator, error) {
-	list := &api.NodeList{}
-	err := factory.Client.Get().Resource("minions").Do().Into(list)
+	allNodes := &api.NodeList{}
+	err := factory.Client.Get().Resource("minions").Do().Into(allNodes)
 	if err != nil {
 		return nil, err
 	}
-	return &nodeEnumerator{list}, nil
+	nodes := &api.NodeList{
+		TypeMeta: allNodes.TypeMeta,
+		ListMeta: allNodes.ListMeta,
+	}
+	for _, node := range allNodes.Items {
+		if len(node.Status.Conditions) == 0 {
+			// If no condition is set, node health check is disabled (master flag
+			// healthCheckMinions is set to false), we add nodes unconditionally.
+			glog.Infof("No condition set for node %v", node.Name)
+			nodes.Items = append(nodes.Items, node)
+			continue
+		}
+		for _, condition := range node.Status.Conditions {
+			// TODO: Distinguish different conditions and implement better condition check.
+			switch condition.Kind {
+			case api.NodeReady:
+				if condition.Status == api.ConditionFull {
+					nodes.Items = append(nodes.Items, node)
+				} else {
+					glog.V(2).Infof("Ignore unhealthy node %v", node.Name)
+				}
+			default:
+				glog.Errorf("Unknown condition kind %v for node %v", condition.Kind, node.Name)
```

Comment:
OK, sorry to be a pain here, but let me suggest this procedure instead:

1. First, look for NodeReady. If that's found, use what it says.
2. If that's not present, then look for NodeReachable. If that's found, use what it says.
3. Otherwise, assume it's healthy.

Extra statuses are not error conditions and shouldn't have a log message, IMO.